graph/product/product_example_test.go

```diff
@@ -128,10 +129,6 @@ func ExampleModular_subgraphIsomorphism() {
 	//     C    4    C    4
 }
 
-// byLength implements the sort.Interface, sorting the slices
-// descending by length.
-type byLength [][]graph.Node
-
-func (n byLength) Len() int           { return len(n) }
-func (n byLength) Less(i, j int) bool { return len(n[i]) > len(n[j]) }
-func (n byLength) Swap(i, j int)      { n[i], n[j] = n[j], n[i] }
+func sortByLengthDescending(mc [][]graph.Node) {
+	slices.SortFunc(mc, func(a, b []graph.Node) int { return cmp.Compare(len(b), len(a)) })
```

Comment:
So this is borderline. The wrapping function name makes it better, but the clarity of a `<` or `>` operator in the condition gives a clear indication of the sort behaviour. In this approach, the order of the `a` and `b` labels is the indications. I think this can be missed (I did until I was looking again at the force push diff). I'm not sure what to do here. This is in testing, so clarity is a higher priority than performance. I would suggest either using `sort.Slice` or `slices.SortFunc` with `-cmp.Compar`, though lean towards `sort.Slice` for reduced page weight. This comment applies to other cases where the sort order is surprising.

Reply:
Okay, that makes sense. This is the only case I could find where the sort order was surprising, and I've decided to use `sort.Slice` as per your preferences.